pkg/core/block/header.go

```diff
@@ -99,6 +99,16 @@ func (b *Header) Hash() util.Uint256 {
 	return b.hash
 }
 
+// SetHash updates Header hash. This method directly modifies Header's hash
+// without any respect to the Header's content. The resulting hash is being
+// cached, hence subsequent calls to Hash will return exactly the same value
+// without an attempt to serialize Header. This method is designated for
+// `invokecontainedscript` RPC API users only and MUST NOT be used for other
+// purposes.
+func (b *Header) SetHash(hash util.Uint256) {
```

Comment:
But why `UnmarshalJSONUnsafe` isn't enough? You've got the hash there anyway.

Reply:
Removed for block since block hash mocking functionality is useless. But we still have to keep it for transaction, ref. https://github.com/nspcc-dev/neo-go/pull/3839#discussion_r2002705013 unless you have some better solution.